OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -107,10 +107,10 @@ public interface INotifyPassengerExited { void OnPassengerExited(Actor self, Act
 	[RequireExplicitImplementation]
 	public interface IConditionConsumerInfo : ITraitInfo { }
 
-	public interface IConditionConsumer
+	public delegate void ConditionConsumer(Actor self, IReadOnlyDictionary<string, int> conditions);
+	public interface IConditionConsumerProvider
 	{
-		IEnumerable<string> Conditions { get; }
-		void ConditionsChanged(Actor self, IReadOnlyDictionary<string, int> conditions);
+		IEnumerable<Pair<ConditionConsumer, IEnumerable<string>>> GetConsumersWithTheirConditions();
```

Comment:
I would prefer that we didn't have to explicitly use `Pair<ConditionConsumer, IEnumerable<string>>`'s everywhere, so could we pull out a specific struct for this?